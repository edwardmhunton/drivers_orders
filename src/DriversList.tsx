import React from "react";

import { Driver } from "./Driver.tsx";

const styles = {
  display: "flex",
  flexDirection: "column"
};

export class DriversList extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { drivers } = this.props;
    return (
      <div style={styles} className="driverlist">
        {drivers.map((item, index) => {
          return <Driver key={index} driver={item} />;
        })}
      </div>
    );
  }
}
