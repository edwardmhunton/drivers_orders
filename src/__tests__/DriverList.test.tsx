import React from "react";

import renderer from "react-test-renderer";

import { DriversList } from "../DriversList";

const driversData = {
  drivers: [
    {
      id: "9823983445",
      coordinates: { lat: "51.54473", lng: " -0.1357" },
      proximity: false,
      distance: 0
    },
    {
      id: "8376587365",
      coordinates: { lat: "51.57131", lng: "-0.1048" },
      proximity: false,
      distance: 0
    },
    {
      id: "456457568",
      coordinates: { lat: "51.57269", lng: "-0.03923" },
      proximity: false,
      distance: 0
    }
  ]
};

describe("DriverList test", () => {
  it("Renders Correctly", () => {
    const DriverListComponent = renderer
      .create(<DriversList drivers={driversData.drivers} />)
      .toJSON();
    expect(DriverListComponent).toMatchSnapshot();
  });
});
