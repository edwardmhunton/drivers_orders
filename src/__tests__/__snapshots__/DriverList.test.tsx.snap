// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriverList test Renders Correctly 1`] = `
<div
  className="driverlist"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "marginLeft": 50,
    }
  }
>
  <div
    className="badge badge-secondary"
    style={
      Object {
        "color": "red",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 20,
        "height": 50,
        "margin": 10,
        "textAlign": "left",
        "width": 300,
      }
    }
  >
    Driver ID: 9823983445
  </div>
  <div
    className="badge badge-secondary"
    style={
      Object {
        "color": "red",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 20,
        "height": 50,
        "margin": 10,
        "textAlign": "left",
        "width": 300,
      }
    }
  >
    Driver ID: 8376587365
  </div>
  <div
    className="badge badge-secondary"
    style={
      Object {
        "color": "red",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 20,
        "height": 50,
        "margin": 10,
        "textAlign": "left",
        "width": 300,
      }
    }
  >
    Driver ID: 456457568
  </div>
</div>
`;
